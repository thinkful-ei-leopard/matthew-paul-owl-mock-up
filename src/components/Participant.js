import React from 'react';

function Participant(props) {
  let sessionClass = props.inSession ? 'online' : 'offline';
  let stageClass = props.onStage ? 'onStage' : 'hidden';
  return (
    <div className="Participant">
      <img className="Participant-Avatar"  src={props.avatar} alt="Avatar" />
      <span className="Participant-Name">{props.name}</span>
      <div className={sessionClass}></div>
      <span className={stageClass}></span>
    </div>
  );
}

export default Participant;