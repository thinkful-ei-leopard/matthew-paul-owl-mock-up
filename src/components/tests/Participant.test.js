import React from 'react';
import ReactDOM from 'react-dom';
import Participant from '../Participant';
import renderer from 'react-test-renderer'; 

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Participant key='test1'
  name='test'
  avatar='https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1'
  inSession={true}
  onStage={true}/>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
  .create(<Participant key='test2'
      name='test'
      avatar='test'
      inSession={true}
      onStage={true}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });