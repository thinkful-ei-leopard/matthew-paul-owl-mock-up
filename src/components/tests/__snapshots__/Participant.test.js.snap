// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Participant"
>
  <img
    alt="Avatar"
    className="Participant-Avatar"
    src="test"
  />
  <span
    className="Participant-Name"
  >
    test
  </span>
  <div
    className="online"
  />
  <span
    className="onStage"
  />
</div>
`;
